capsules/detector_text_openvino/backend.py

```diff
@@ -0,0 +1,116 @@
+class OpenVINOModel(BaseOpenVINOBackend):
+    def process_frame(self,
+                      frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) -> DETECTION_NODE_TYPE:
+        raise NotImplemented('we no use this lel')
+
+
+class Backend(BaseBackend):
+    label_map: Dict[int, str] = {1: "text"}
+
+    def __init__(self, detector: OpenVINOModel,
+                 recognizer_encoder: OpenVINOModel,
+                 recognizer_decoder: OpenVINOModel):
+        super().__init__()
+        self.detector = detector
+        self.recognizer_encoder = recognizer_encoder
+        self.recognizer_decoder = recognizer_decoder
+
+    @property
+    def workload(self) -> float:
+        return (self.detector.workload +
+                self.recognizer_encoder.workload +
+                self.recognizer_decoder.workload)
+
+    def batch_predict(self, input_data_list: List[Any]) -> List[Any]:
+        raise NotImplemented('dont use this lel')
+
+    def process_frame(self, frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) -> DETECTION_NODE_TYPE:
+        n, c, h, w = self.detector.net.inputs['im_data'].shape
+        hidden_shape = self.recognizer_decoder.net.inputs['prev_hidden'].shape
+
+        input_dict, resize = self.detector.prepare_inputs(
+            frame,
+            frame_input_name="im_data"
+        )
+        input_dict["im_data"] = (input_dict["im_data"]
+                                 .reshape((n, c, h, w)).astype(np.float32))
+
+        input_image_size = self.detector.net.inputs['im_data'].shape[-2:]
+        input_image_info = np.asarray(
+            [[input_image_size[0], input_image_size[1], 1]], dtype=np.float32)
+        input_dict["im_info"] = input_image_info
+        prediction = self.detector.send_to_batch(input_dict).get()
+
+        scores = prediction["scores"]
+        detections_filter = scores > options["threshold"]
+        scores = scores[detections_filter]
+        rects = prediction["boxes"][detections_filter]
+        text_features = prediction["text_features"][detections_filter]
+
+        feature_queues = []
+        for text_feature in text_features:
+            feature_queues.append(
+                self.recognizer_encoder.send_to_batch({'input': text_feature}))
+
+        detections = []
+        for score, rect, feature_queue in zip(scores, rects, feature_queues):
+            feature = feature_queue.get()['output']
+            feature = np.reshape(feature,
+                                 (feature.shape[0], feature.shape[1], -1))
+            feature = np.transpose(feature, (0, 2, 1))
+
+            hidden = np.zeros(hidden_shape)
+            prev_symbol_index = np.ones((1,)) * SOS_INDEX
```

Comment:
Why not just `np.asarray([SOS_INDEX])`?

Reply:
this is how openVINO did it in their code. is your version identical?

Reply:
It _should_ be. I have no idea why OpenVINO would have done that